docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,454 @@
+   Even though this requirement is realistic, thin-edge MQTT topics must not be corrupted with cloud specific semantics,
+   and the same requirement will be handled with some external routing mechanism(e.g: routing via bridge topics)
+
+## Example deployment
+
+Here is a sample deployment structure for a smart two-wheeler that is used in the examples below:
+
+```mermaid
+graph TD;
+    TEDGE --> CENTRAL-ECU-0001;
+    CENTRAL-ECU-0001 --> ENGINE-ECU-E001;
+    CENTRAL-ECU-0001 --> WHEEL-ECU-E001;
+    CENTRAL-ECU-0001 --> WHEEL-ECU-E002;
+
+    ENGINE-ECU-E001 --> TMP-0001((TMP-0001));
+
+    WHEEL-ECU-E001 --> BRAKE-ECU-A001;
+    WHEEL-ECU-E001 --> TYRE-ECU-T001;
+    BRAKE-ECU-A001 --> TMP-1001((TMP-0001));
+    TYRE-ECU-T001 --> TMP-1002((TMP-0002));
+    TYRE-ECU-T001 --> PRSR-1002((PRSR-0002));
+
+    WHEEL-ECU-E002 --> BRAKE-ECU-A001'[BRAKE-ECU-A001];
+    WHEEL-ECU-E002 --> TYRE-ECU-T002;
+    BRAKE-ECU-A001' --> TMP-1001'((TMP-0001));
+    TYRE-ECU-T002 --> TMP-1002'((TMP-0002));
+    TYRE-ECU-T002 --> PRSR-1002'((PRSR-0002));
+```
+
+As you can see, the ECUs for the front and rear wheels have unique IDs: `WHEEL-ECU-E001` and `WHEEL-ECU-E002`,
+as they exist at the same level, connected to the central ECU(`CENTRAL-ECU-0001`).
+But, the brake ECUs connected to both the wheels may have the same ID, as they are not linked directly anyway.
+Even the sensors attached at many levels in such a complex deployment may have the same IDs(`TMP-0001`).
+
+So, the proposed solutions must address such ID clashes in a deep nested hierarchies.
+
+## Proposals
+
+### Proposal 1: Dedicated topics for devices and services
+
+This proposal is built on the following assumptions/constraints:
+* thin-edge can identify the child devices(the nested ones as well), and all the services with unique IDs.
+* All child-devices connected to a parent device will have unique IDs under its parent namespace,
+  but not necessarily across the entire tedge namespace/hierarchy.
+* If all child devices **can not** guarantee that uniqueness across the entire hierarchy,
+  then they are expected to register with thin-edge to get that unique ID generated.
+* Given that unique ID, thin-edge can identify its entire parent lineage and descendent child devices.
+  Thin-edge internally maintains this mapping information as child devices register with it.
+* If all child devices **can** guarantee ID uniqueness across the entire hierarchy,
+  they can skip this registration step.
+* Services need not have unique IDs across the entire hierarchy.
+  They just need to have unique IDs under their parent's namespace.
+
+Topics to have device id as the top level prefix with distinction on the target: device or service as in:
+`tedge/<device-id>/<target-type>/...`
+
+For tedge device: tedge/main/...
+For tedge device services: tedge/main/service/<service-id>/...
+For child devices: tedge/<child-id>/...
+For child device services: tedge/<child-id>/service/<service-id>/...
```

Comment:
There was an alternate thought to keep the topics of the main/child devices symmetric to that of the service by having a scheme like `tedge/<device-id>/<dummy-placeholder>/<dummy-placeholder>/` with 4 levels as in the case of services.

That would have enabled one-line subscriptions for cases like "all measurements from all devices and services" with just `tedge/+/+/+/telemetry/measurements`. But decided against it to keep the main device topics simples as the same kind of query could be achieved with just two topics instead of one.